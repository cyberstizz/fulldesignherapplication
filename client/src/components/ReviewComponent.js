import React from 'react';
import './ReviewComponent.scss';

const ReviewComponent = ({ headline, rating, review, userName }) => {
  return (
    <div className="reviewComponent">
      {/* Display the review headline */}
      <h3 className="headline">{headline}</h3>
      {/* Display the rating */}
      <div className="rating">Rating: {rating}</div>
      {/* Display the review text */}
      <p className="review">{review}</p>
      {/* Display the reviewer's username */}
      <p className="userName">- {userName}</p>
    </div>
  );
};

export default ReviewComponent;
