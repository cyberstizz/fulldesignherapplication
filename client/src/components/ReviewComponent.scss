.reviewComponent {
    background: linear-gradient(to right, #FF0000, #CC0000, #990000, #660000);
    padding: 20px;
    margin-bottom: 20px;
    border-radius: 10px;
    color: black;
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
    
  
    .headline {
      margin: 0;
      padding-bottom: 10px;
      border-bottom: 1px solid #fff;
      color: black
    }
  
    .rating,
    .review,
    .userName {
      margin-top: 10px;
      color: black
    }
  }
  