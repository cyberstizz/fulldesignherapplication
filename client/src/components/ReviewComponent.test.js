// src/components/ReviewComponent.test.js
import React from 'react';
import { render, screen } from '@testing-library/react';
import ReviewComponent from './ReviewComponent';

describe('ReviewComponent', () => {
  const testProps = {
    headline: 'Great Product!',
    rating: 5,
    review: 'I absolutely loved this product. Highly recommend!',
    userName: 'John Doe',
  };

  test('renders headline correctly', () => {
    render(<ReviewComponent {...testProps} />);

    expect(screen.getByText(testProps.headline)).toBeInTheDocument();
  });

  test('renders rating correctly', () => {
    render(<ReviewComponent {...testProps} />);

    expect(screen.getByText(`Rating: ${testProps.rating}`)).toBeInTheDocument();
  });

  test('renders review text correctly', () => {
    render(<ReviewComponent {...testProps} />);

    expect(screen.getByText(testProps.review)).toBeInTheDocument();
  });

  test('renders user name correctly', () => {
    render(<ReviewComponent {...testProps} />);

    expect(screen.getByText(`- ${testProps.userName}`)).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<ReviewComponent {...testProps} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
