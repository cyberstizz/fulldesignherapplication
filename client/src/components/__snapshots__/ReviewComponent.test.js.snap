// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewComponent matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="reviewComponent"
  >
    <h3
      class="headline"
    >
      Great Product!
    </h3>
    <div
      class="rating"
    >
      Rating: 5
    </div>
    <p
      class="review"
    >
      I absolutely loved this product. Highly recommend!
    </p>
    <p
      class="userName"
    >
      - John Doe
    </p>
  </div>
</DocumentFragment>
`;
